### ISSUE TITLE
Add Support for Python 3.13

### ISSUE THREAD

**Issue by pmbarrett314** [zephyrproject-rtos/windows-curses#69]
Python 3.13 is out, which means this project needs an update to support it. I gave it [the old college try](https://github.com/pmbarrett314/windows-curses), but ran into [some snags](https://github.com/pmbarrett314/windows-curses/actions/runs/11338298421/job/31531276743) I don't have the bandwidth for. Creating an issue if anyone wants to point out anything glaring I've done wrong, or failing that just to track the status of support for the new version of Python.

**Comment by DjLegolas** [comment]
I have checked and found the problem.
There is no default `curses.h` to be included - every option has its own macro.
In addition, this is the same for including `panel.h` in `_curses_panel.c` that you have added in py3.13 folder.

I have opened a [MR](https://github.com/pmbarrett314/windows-curses/pull/1) in your fork so you can use it and change whatever you need before opening a MR here.

**Comment by MEILcodes** [comment]
i have also ran into this problem when i try to follow getting started guide of zephyr OS. when running "pip install -r ~/zephyrproject/zephyr/scripts/requirements.txt" for  installing python dependencies. it will say "ERROR: Could not find a version that satisfies the requirement windows-curses (from versions: none)". since i have installed python 3.13 this may be the problem.

**Comment by rahaaatul** [comment]
I get this error while installing:

```powershell
pwsh> pip install --upgrade windows-curses

ERROR: Could not find a version that satisfies the requirement windows-curses (from versions: none)
ERROR: No matching distribution found for windows-curses
```

**Comment by pmbarrett314** [comment]
It looks like https://github.com/python/cpython/issues/110964 is causing issues. I jankily copied and pasted the relevant definitions from the private header in cpython into my fork and things function now, but it is still exiting with a 0xc0000005 that isn't happening in 3.12.

**Comment by stephanosio** [comment]
v2.4.1a1 has been released with Python 3.13 support for initial testing.

**Comment by LewisNeal** [comment]
Thanks for the update. I have been using windows-curses v2.4.1a1 and Python 3.13.1 successfully with no errors.
